src/main/java/net/minecraftforge/fluids/BlockFluidBase.java

```diff
@@ -492,6 +499,25 @@ public IBlockState getExtendedState(@Nonnull IBlockState oldState, @Nonnull IBlo
                 }
             }
         }
+        
+        //check for downflow above corners
+        if(isCoveredWithFluid(worldIn, pos.add(-1, 0, -1)) || isCoveredWithFluid(worldIn, pos.add(-1, 0, 0)) || isCoveredWithFluid(worldIn, pos.add(0, 0, -1)))
+        {
+            corner[0][0] = 1;
+        }
+        if(isCoveredWithFluid(worldIn, pos.add(-1, 0, 0)) || isCoveredWithFluid(worldIn, pos.add(-1, 0, 1)) || isCoveredWithFluid(worldIn, pos.add(0, 0, 1)))
+        {
+            corner[0][1] = 1;
+        }
+        if(isCoveredWithFluid(worldIn, pos.add(0, 0, -1)) || isCoveredWithFluid(worldIn, pos.add(1, 0, -1)) || isCoveredWithFluid(worldIn, pos.add(1, 0, 0)))
+        {
+            corner[1][0] = 1;
+        }
+        if(isCoveredWithFluid(worldIn, pos.add(1, 0, 0)) || isCoveredWithFluid(worldIn, pos.add(0, 0, 1)) || isCoveredWithFluid(worldIn, pos.add(1, 0, 1)))
+        {
+            corner[1][1] = 1;
+        }
+        
```

Comment:
![neighbour](https://cloud.githubusercontent.com/assets/10445063/23509330/a3c4f29e-ff55-11e6-874b-441bbca22991.png)

The reason I'm testing for neighbours is becuase of situations like this. But you are right, as you can see in the screenshot, vanilla does not check for it. I would argue checking for it is actually more correct behaviour. But I also understand if it's more important to retain compatibility to vanilla mechanics. Should I get rid of the check?